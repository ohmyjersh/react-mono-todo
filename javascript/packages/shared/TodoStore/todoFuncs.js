import v4 from "uuid";

export const toggleTodo = (state, id) => ({
    todos: state.todos.map(todo => {
        if (todo.id === id) {
            return { ...todo,
                completed: !todo.completed
            }
        }
        return todo
    }),
});

export const clearTodo = (state, id) => ({
    todos: state.todos.filter(todo => todo.id !== id),
});

export const addTodo = (state, title) => ({
    todos: state.todos.concat({
        id: v4(),
        title,
        completed: false
    }),
});