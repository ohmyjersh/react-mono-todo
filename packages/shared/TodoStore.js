import React from "react"
import v4 from "uuid";

const defaultState = {
  error: null,
  loading: false,
  todos: [],
}

export const TodosContext = React.createContext(defaultState)

export default class TodosStore extends React.Component {
  state = defaultState


  toggleTodo = id => {
      this.setState(state => ({
        todos: state.todos.map(todo => {
          if (todo.id === id) {
            return { ...todo, completed:!todo.completed}
          }
          return todo
        }),
      }));
  }

  clearTodo = id => {
      this.setState(state => ({
        todos: state.todos.filter(todo => todo.id !== id),
      }));
 }

  addTodo = title => {
      this.setState(state => ({
        todos: state.todos.concat({id:v4(), title, completed:false}),
      }));
}

  render() {
    return (
      <TodosContext.Provider
        value={{
          error: this.state.error,
          loading: this.state.loading,
          todos: this.state.todos,
          loadTodos: this.loadTodos,
          toggleTodo: this.toggleTodo,
          clearTodo: this.clearTodo,
          addTodo: this.addTodo,
        }}>
        {this.props.children}
      </TodosContext.Provider>
    )
  } 
}