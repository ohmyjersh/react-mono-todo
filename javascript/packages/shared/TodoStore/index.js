import React from "react"
import * as todoFuncs from './todoFuncs';

const defaultState = {
  error: null,
  loading: false,
  todos: [{id:0, title:'shared', completed:false}],
}

export const TodosContext = React.createContext(defaultState)

export default class TodosStore extends React.Component {
  state = defaultState


  toggleTodo = id => {
      this.setState(state => todoFuncs.toggleTodo(state, id));
  }

  clearTodo = id => {
      this.setState(state => todoFuncs.clearTodo(state, id));
 }

  addTodo = title => {
      this.setState(state => todoFuncs.addTodo(state, title));
}

  render() {
    return (
      <TodosContext.Provider
        value={{
          stuff:'hi',
          error: this.state.error,
          loading: this.state.loading,
          todos: this.state.todos,
          loadTodos: this.loadTodos,
          toggleTodo: this.toggleTodo,
          clearTodo: this.clearTodo,
          addTodo: this.addTodo,
        }}>
        {this.props.children}
      </TodosContext.Provider>
    )
  } 
}