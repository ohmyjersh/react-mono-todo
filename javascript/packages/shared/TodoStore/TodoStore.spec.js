import React from 'react';
import TodoStore from './index';
import renderer from 'react-test-renderer';

test('should render the TodoStore', () => {
    const tree = renderer
    .create(<TodoStore />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});