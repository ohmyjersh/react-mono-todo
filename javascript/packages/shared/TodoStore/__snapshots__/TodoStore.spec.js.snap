// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the TodoStore 1`] = `null`;
